### DISCUSSION TITLE
有没有什么方法通过Base64，去人脸库里查询，并且返回识别度最高的imagesId出来

### DISCUSSION THREAD

**Discussion by yang1158280920** [exadel-inc/CompreFace#1037] [Q&A]
目前想通过Base64字符串，调用接口，拿到识别度最高的imagesId出来

**Comment by yang1158280920** [comment]
目前的场景是，我通过摄像头捕捉人脸，然后想通过这个人脸知道这个人是谁。我目前的方法是通过拿到的人脸Base64，去存储了人脸imagesId，通过Base64+imagesId去循环比对。拿到识别度最高的那个imagesId  然后就反查到了这个人是谁。但是现在这个方法影响性能，响应的速度太慢了。想问问有没有什么更好的方法去 通过Base64 直接能拿到识别度最高的imagesId

**Comment by yang1158280920** [comment]
或者是有没有  一个Base64同时比较imagesId集合或者数组。然后把识别度最高的imagesId返回出来的方法

**Comment by pospielov** [comment]
Not sure I understood the case. I think this is a base face recognition case when you upload known faces first and then find the most similar ones. 
You need to use [recognition](https://github.com/exadel-inc/CompreFace/blob/master/docs/Rest-API-description.md#recognize-faces-from-a-given-image) endpoint
You need to set `prediction_count` to `-1`
Then it will return similarities to all faces in the face collection.

**Reply by yang1158280920** [reply]
![image](https://user-images.githubusercontent.com/61954441/223309615-28ba3940-7bd1-4980-8834-fa26805f61fa.png)

I use this API     

I mainly want to get the corresponding imagesId through Base64

**Reply by yang1158280920** [reply]
The scene I want is a room with 30 people. I want to recognize who is coming in through face recognition

**Reply by pospielov** [reply]
I recommend using this endpoint instead:
https://github.com/exadel-inc/CompreFace/blob/master/docs/Rest-API-description.md#recognize-faces-from-a-given-image
It should do what you need.
